### PR TITLE
[WIP] migrate API v2

### PR DESCRIPTION
**Pull request by s-aska**
- [x] ls
- [x] find
- [x] copy
- [x] move
- [x] mkdir
- [x] delete
- [x] upload
- [x] download
- [x] sync
- [x] help

### CODE REVIEW

File: script/dropbox-api [hunk 1]
```diff
@@ -65,35 +69,54 @@ my $config = decode_json($config_file->slurp);
...
+        $config_file->openw->print(encode_json($config));
+        warn "=> Suucess.";
+    } else {
+        die "please setup.";
+    }
+}
+if (my $access_level = delete $config->{access_level}) {
     if ($access_level eq 'a') {
-        $context = 'sandbox';
+        print "sandbox is gone, Are you sure you want to delete from the config the access_leve? [y/n]: ";
```
Comment (mathstuf) [inline script/dropbox-api]: "access_leve" -> "access_level"

File: script/dropbox-api [hunk 2]
```diff
@@ -65,35 +69,54 @@ my $config = decode_json($config_file->slurp);
...
+        chomp( my $y = <STDIN> );
+        if ($y =~ qr{ [yY] }xms) {
+            delete $config->{access_level};
+            $config_file->openw->print(encode_json($config));
+            warn "=> Suucess.";
+        } else {
+            die "canceled.";
```
Comment (mathstuf) [inline script/dropbox-api]: "cancelled"

File: script/dropbox-api [hunk 3]
```diff
@@ -356,30 +366,19 @@ sub setup {
...
-    my $login_link = $box->login;
+    my $login_link = $box->authorize;
     die $box->error if $box->error;
     print "URL: $login_link\n";
-    print "Please Access URL and press Enter\n";
-    print "OK?";
-    <STDIN>;
-    $box->auth;
-    die $box->error if $box->error;
+    print "Please Access URL\n";
```
Comment (mathstuf) [inline script/dropbox-api]: "please" -> "missing"?

File: lib/App/dropboxapi.pm [hunk 4]
```diff
@@ -17,7 +17,85 @@ Run C<dropbox-api help> for more options.
...
+
+    > dropbox-api setup
+    Please Input API Key: ***************
+    Please Input API Secret: ***************
+    Please Input Access type
+      a ... App folder - Your app only needs access to a single folder within the user's Dropbox
+      f ... Full Dropbox - Your app needs access to the user's entire Dropbox
+    [a or f]: *
+    URL: http://api.dropbox.com/0/oauth/authorize?oauth_token=***************&oauth_callback=
+    Please Access URL and press Enter
```
Comment (mathstuf) [inline lib/App/dropboxapi.pm]: This doesn't match what is in the code anymore.

File: lib/App/dropboxapi.pm [hunk 5]
```diff
@@ -49,7 +127,11 @@ dropbox-api help [<command>]
         put   upload file
         get   download file
         sync  sync directory (local => dropbox or dropbox => local)
-        uid   get accound uid
+
+    Common Options
+        -e enable env_proxy ( HTTP_PROXY, NO_PROXY )
+        -D enable debug
+        -v verbose
```
Comment (mathstuf) [inline lib/App/dropboxapi.pm]: Is -s documented here? It should be removed since it is no longer actually supported.

**Comment by mathstuf** [comment]
Thanks! Did a once-over for user-facing strings.

**Comment by s-aska** [comment]
Very Thank you, was corrected immediately.
https://github.com/s-aska/dropbox-api-command/commit/352a3ffd70996aae6dbc42caaaf07b0a7fee3338

**Pull request merged**